websites/M/movie-web/presence.ts

```diff
@@ -0,0 +1,294 @@
+
+			if ((state.progress.time && state.progress.duration) !== 0) {
+				presenceData.state = createProgressBar(
+					state.progress.time,
+					state.progress.duration,
+					{
+						barLengthString,
+						barFill,
+						barTrack,
+						showLabel,
+					}
+				);
+			}
+
+			const title = `${media.meta.meta.title} (${media.meta.meta.year})`;
+			if (media.meta.meta.type === "series") {
+				const episodeData = media.episode,
+					episode = media.meta.meta.seasonData.episodes.find(
+						episode => episode.id === episodeData.episodeId
+					);
+
+				presenceData.details = `S${media.meta.meta.seasonData.number}E${episode.number} — ${title}`;
+			} else presenceData.details = title;
+
+			if (state.mediaPlaying.isFirstLoading) {
+				presenceData.smallImageKey = "https://i.imgur.com/lBCZWIB.gif";
+				presenceData.smallImageText = "Loading";
+			} else if (state.mediaPlaying.isPlaying) {
+				const now = new Date();
+				presenceData.endTimestamp = now.setSeconds(
+					now.getSeconds() + state.progress.duration - state.progress.time
+				);
+				presenceData.smallImageKey = Assets.Play;
+				presenceData.smallImageText = "Playing";
+			} else if (state.mediaPlaying.isPaused) {
+				presenceData.smallImageKey = Assets.Pause;
+				presenceData.smallImageText = "Paused";
+			}
+		}
+	} else if (subdomain) {
+		if (pathname === "" || pathname.startsWith("/search"))
+			presenceData.startTimestamp = browsingTimestamp;
+		else if (pathname.startsWith("/media")) {
+			const metaObj = await presence.getPageletiable("meta"),
+				metaData: ProdMetaData | undefined = Object.values(metaObj ?? {})[0];
+			if (!metaData) return;
+
+			const { progress, meta, episode } = metaData,
+				mediaPlaying = {
+					isFirstLoading: (progress.time && progress.duration) === 0,
+					isLoading: progress.duration === 0,
+				};
+
+			presenceData.largeImageKey = meta.meta.poster;
+
+			if (!mediaPlaying.isLoading && progress.time && progress.duration) {
+				presenceData.state = createProgressBar(
+					progress.time,
+					progress.duration,
+					{
+						barLengthString,
+						barFill,
+						barTrack,
+						showLabel,
+					}
+				);
+			}
+
+			const title = `${meta.meta.title} (${meta.meta.year})`;
+			if (meta.meta.type === "series" && episode) {
+				presenceData.details = `S${meta.meta.seasonData.number}E${
+					meta.meta.seasonData.episodes.find(
+						episode => episode.id === episode.id
+					).number
+				} — ${title}`;
+			} else presenceData.details = title;
+
+			if (mediaPlaying.isFirstLoading) {
+				presenceData.smallImageKey = "https://i.imgur.com/lBCZWIB.gif";
+				presenceData.smallImageText = "Loading";
+			} else {
+				const now = new Date();
+				presenceData.endTimestamp = now.setSeconds(
+					now.getSeconds() + progress.duration - progress.time
+				);
+			}
+		}
+	}
+
+	if (showWatchButton && !pathname.startsWith("/search")) {
+		presenceData.buttons = [
+			{
+				label: "Watch",
```

Comment:
```suggestion
				label: "Watch Video",
```

Reply:
Can't we keep it to just "Watch"?

Reply:
> Can't we keep it to just "Watch"?

We typically use Watch Video if its about a video

Reply:
You can probs do something like:
```ts
`Watch ${meta.meta.type === "series" ? "Series" : "Video"}`
```